dev/setup/shell.php

```diff
@@ -63,15 +63,35 @@ function get_terminal_lines_cols() {
 			$columns = $m['val'];
 		}
 	} else {
-		$tput_exists = ! empty( exec( "command -v tput" ) );
+		// Lines
+		exec( 'tput lines', $lines, $status );
 
-		if ( ! $tput_exists ) {
-			// We can't get the lines and columns on this terminal, bail.
+		if ( $status !== 0 ) {
+			// tput lines can't give us the lines on this terminal, bail.
 			return [ $lines, $columns ];
 		}
 
-		$lines   = abs( intval( exec( 'tput lines' ) ) );
-		$columns = abs( intval( exec( 'tput cols' ) ) );
+		if ( ! empty( $lines ) && is_numeric( $lines[0] ) ) {
```

Comment:
This code is verbose, it could be slimmed down a bit after the `$columns` variable is renamed to `$cols`:

```php
foreach( ['lines', 'cols'] as $key ){
	exec( 'tput ' . $key, $output, $status );
	if( $status !== 0 || empty( $output[0] ){
		return [$lines, $columns];
	}
	$$key = max($$key, abs( (int)$output[0] ));
}

return [$lines, $columns];
```

Reply:
Less LOC. Harder to read. Not sure which is best, but I trust you if you'd like to take this path.

- https://johnkary.net/slides/loops/midwest-php-2016/images/comic1.png
- https://johnkary.net/slides/loops/midwest-php-2016/images/comic2.png
- https://johnkary.net/slides/loops/midwest-php-2016/images/comic3.png

Reply:
it comes down to personal taste I guess, I do not perceive it as hard to read, but it's fine either way.

Reply:
https://i.makeagif.com/media/10-05-2015/M_HY8c.gif